packages/library-base/src/commonMain/kotlin/io/realm/kotlin/serializers/RealmKSerializers.kt

```diff
@@ -0,0 +1,498 @@
+ * @file:UseSerializers(
+ *     RealmListKSerializer::class,
+ *     RealmSetKSerializer::class,
+ *     RealmAnyKSerializer::class,
+ *     RealmInstantKSerializer::class,
+ *     MutableRealmIntKSerializer::class,
+ *     RealmUUIDKSerializer::class
+ * )
+ * ```
+ *
+ * Serializers for all Realm data types can be found in [io.realm.kotlin.serializers].
+ */
+public class RealmSetKSerializer<E>(elementSerializer: KSerializer<E>) : KSerializer<RealmSet<E>> {
+    private val serializer = SetSerializer(elementSerializer)
+
+    override val descriptor: SerialDescriptor =
+        serializer.descriptor
+
+    override fun deserialize(decoder: Decoder): RealmSet<E> =
+        serializer.deserialize(decoder).toRealmSet()
+
+    override fun serialize(encoder: Encoder, value: RealmSet<E>) {
+        serializer.serialize(encoder, value)
+    }
+}
+
+/**
+ * KSerializer implementation for [RealmDictionary]. Serialization is done as a generic map structure,
+ * whilst deserialization is done into an unmanaged [RealmDictionary].
+ *
+ * It supports any serializable type as a type argument.
+ *
+ * The serializer must be registered per property:
+ * ```
+ * class Example : RealmObject {
+ *     @Serializable(RealmDictionaryKSerializer::class)
+ *     var myDictionary: RealmDictionary<String> = realmDictionaryOf()
+ * }
+ * ```
+ * or per file:
+ * ```
+ * @file:UseSerializers(RealmDictionaryKSerializer::class)
+ *
+ * class Example : RealmObject {
+ *     var myDictionary: RealmDictionary<String> = realmDictionaryOf()
+ * }
+ * ```
+ *
+ * Adding the following code snippet to a Kotlin file would conveniently register any field using a
+ * Realm datatype to its correspondent serializer:
+ *
+ * ```
+ * @file:UseSerializers(
+ *     RealmListKSerializer::class,
+ *     RealmSetKSerializer::class,
+ *     RealmAnyKSerializer::class,
+ *     RealmInstantKSerializer::class,
+ *     MutableRealmIntKSerializer::class,
+ *     RealmUUIDKSerializer::class
+ * )
+ * ```
+ *
+ * Serializers for all Realm data types can be found in [io.realm.kotlin.serializers].
+ */
+public class RealmDictionaryKSerializer<E>(elementSerializer: KSerializer<E>) :
+    KSerializer<RealmDictionary<E>> {
+    private val serializer = MapSerializer(String.serializer(), elementSerializer)
+
+    override val descriptor: SerialDescriptor =
+        serializer.descriptor
+
+    override fun deserialize(decoder: Decoder): RealmDictionary<E> =
+        serializer.deserialize(decoder).toRealmDictionary()
+
+    override fun serialize(encoder: Encoder, value: RealmDictionary<E>) {
+        serializer.serialize(encoder, value)
+    }
+}
+
+/**
+ * KSerializer implementation for [RealmInstant]. It is serialized as a [BsonDateTime], to allow direct
+ * usage on Mongodb function calls, and deserialized as an unmanaged [MutableRealmInt].
+ *
+ * Warning: because [RealmInstant] and [BsonDateTime] have different precision the serialization would
+ * lose precision from nanoseconds to milliseconds.
```

Comment:
```suggestion
 * KSerializer implementation for [RealmInstant]. It is serialized as a [BsonDateTime], to allow direct
 * usage on Mongodb function calls, and deserialized as an unmanaged [RealmInstant].
 *
 * Warning: because [RealmInstant] and [BsonDateTime] have different precision the serialization will
 * lose precision as nanoseconds get truncated to milliseconds.
```